apis/v1alpha1/httproute_types.go

```diff
@@ -145,13 +146,23 @@ type HTTPRouteMatch struct {
 	// +optional
 	// +kubebuilder:validation:Enum=Exact;ImplementationSpecific
 	// +kubebuilder:default=Exact
-	HeaderMatchType *string `json:"headerMatchType" protobuf:"bytes,3,opt,name=headerMatchType"`
+	Type *string `json:"type" protobuf:"bytes,3,opt,name=type"`
 	// Headers are the HTTP Headers to match as interpreted via
 	// HeaderMatchType. Multiple headers are ANDed together, meaning, a request
```

Comment:
Nit: I think HeaderMatchType should also be updated now to reflect the new simplified type.